scripts/generator/graphql_generator/csharp/SDK/Android/AndroidNativeCheckout.cs.erb

```diff
@@ -33,9 +51,16 @@ namespace <%= namespace %>.SDK.Android {
             // TODO
         }
 
-        public bool CanCheckout(PaymentSettings paymentSettings) {
-            // TODO: Invoke methods in the plugin to determine if we can perform Android Pay
-            return true;
+        public void CanCheckout(PaymentSettings paymentSettings, CanCheckoutWithNativePayCallback callback) {
+            if (paymentSettings.supportedDigitalWallets().Contains(DigitalWallet.ANDROID_PAY)) {
```

Comment:
I wonder if we should check to see if we already had a callback set and pass `false` to it before setting another one.

Reply:
The best approach would be to have each callback mapped to a request identifier in order to make sure every callback is called with the right information at the right time just to enforce consistency in case the client decides to have multiple `CanCheckout()` calls running along each other. On the other hand, I thought it'd be overkill for a simple verification that is unlikely to run concurrently. What you suggested ensures no client call will be "callbackless", but the default `false` answer is not necessarily correct. Any thoughts on the overengineered approach I just mentioned?

Reply:
I'd probably keep it simple then. I don't see this being an issue for more developers anyways.

Reply:
https://github.com/Shopify/unity-buy-sdk/pull/360/commits/dd0e537d23c5f6c87b4d1cfb0ac469cecee70a79